### PR TITLE
Jetpack CP: UI for Install Jetpack from App

### PR DESCRIPTION
**Pull request by hafizrahman**
## Description

This is the first part for the "Jetpack CP: Install Jetpack from app" feature: https://github.com/woocommerce/woocommerce-android/issues/4865#issuecomment-956128458

This PR adds the UI elements and navigation related to the functionality. There is no functionality yet and the intent of this PR is just to check the design and test the user interface.

**Exception:** The circular loading animation design is not yet decided, so for now it is using a temporary animation and it can be ignored.

## To test:
1. Have a Jetpack CP site (e.g: start a .org site with WooCommerce, then install WooCommerce Payments plugin, then set it up to connect to your WordPress.com site),
2. Log in to Woo app and select the Jetpack CP site,
3. Find the "Get the full experience with Jetpack" banner on "My Store" tab, then tap it,
4. This will show the Jetpack benefits dialog. Tap "Install Jetpack" button,
5. This will show the install start dialog. Check the design and test it on different orientations,
6. After checking is done, tap "Install Jetpack" button,
7. This will show the install progress dialog. Check the design and test it on different orientations.
8. After checking is done, tap "Done" and make sure it returns to "My Store" screen.

**Light mode**
| Install start | Install progress |
|-|-|
|||

**Dark mode**
| Install start | Install progress |
|-|-|
|||

### CODE REVIEW

File: WooCommerce/src/main/res/layout/dialog_jetpack_install_progress.xml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        android:layout_weight="1"
+        android:fillViewport="true">
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:paddingBottom="@dimen/major_300">
+
+            <LinearLayout
+                android:id="@+id/progress_icon"
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/layout/dialog_jetpack_install_progress.xml]: np, putting the `icon` prefix on a layout seems confusing, I understand why you did it, but still, I would call it `header_layout` or something similar, WDYT?

**Review by hichamboushaba — APPROVED** [review]
Nice work @hafizrahman 💯, I left just one nitpick comment, but feel free to ignore it and merge if you like 🙂

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/JetpackInstallProgressDialog.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package com.woocommerce.android.ui.mystore
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/JetpackInstallProgressDialog.kt]: We need to move those files to their proper package, for example: `com.woocommerce.android.ui.jetpack`, I missed this on my initial review.

**Review by hichamboushaba — CHANGES_REQUESTED** [review]
I just noticed all the files are kept on the `mystore` package, I think they should be moved to their own package.
Even the JetpackBenefitsDialog should've been moved, but I missed it when I worked on this, so if you can move everything to a new package, or at least move the new files.

**Comment by hafizrahman** [comment]
Thank you for your feedbacks @hichamboushaba 

This is ready for another review.

**Review by hichamboushaba — APPROVED** [review]
Thanks for the fixes @hafizrahman 👍 
There is a conflict blocking the merge though

**Comment by hafizrahman** [comment]
Thanks for the approval @hichamboushaba 

I've fixed the conflict, merging now.

**Pull request merged**